babel/dates.py

```diff
@@ -187,23 +211,34 @@ def _get_time(time, tzinfo=None):
 
 def get_timezone(zone=None):
     """Looks up a timezone by name and returns it.  The timezone object
-    returned comes from ``pytz`` and corresponds to the `tzinfo` interface and
-    can be used with all of the functions of Babel that operate with dates.
+    returned comes from ``zoneinfo`` is available or ``pytz`` if not.
+    It corresponds to the `tzinfo` interface and can be used with all of
+    the functions of Babel that operate with dates.
 
     If a timezone is not known a :exc:`LookupError` is raised.  If `zone`
     is ``None`` a local zone object is returned.
 
     :param zone: the name of the timezone to look up.  If a timezone object
-                 itself is passed in, mit's returned unchanged.
+                 itself is passed in, it's returned unchanged.
     """
     if zone is None:
         return LOCALTZ
     if not isinstance(zone, str):
         return zone
-    try:
-        return _pytz.timezone(zone)
-    except _pytz.UnknownTimeZoneError:
-        raise LookupError(f"Unknown timezone {zone}")
+
+    if zoneinfo:
+        try:
+            return zoneinfo.ZoneInfo(zone)
+        except zoneinfo.ZoneInfoNotFoundError:
+            pass
+
+    else:
+        try:
+            return _pytz.timezone(zone)
+        except _pytz.UnknownTimeZoneError:
+            pass
+
+    raise LookupError(f"Unknown timezone {zone}")
```

Comment:
In accordance with the above "support both pytz and zoneinfo" thing, this should prefer pytz... and we might as well keep track of the original exception:
```suggestion
    exc = None
    if pytz:
        try:
            return pytz.timezone(zone)
        except pytz.UnknownTimeZoneError as exc:
            pass

    if zoneinfo:
        try:
            return zoneinfo.ZoneInfo(zone)
        except zoneinfo.ZoneInfoNotFoundError as exc:
            pass

    raise LookupError(f"Unknown timezone {zone}") from exc
```